code/controllers/subsystem/mapping.dm

```diff
@@ -271,7 +274,7 @@ SUBSYSTEM_DEF(mapping)
 
 	var/list/ice_ruins = levels_by_trait(ZTRAIT_ICE_RUINS)
 	for (var/ice_z in ice_ruins)
-		spawn_rivers(ice_z, 4, /turf/open/openspace/icemoon, /area/icemoon/surface/outdoors/unexplored/rivers)
+		spawn_rivers(ice_z, GLOB.ice_river_count, /turf/open/openspace/icemoon, /area/icemoon/surface/outdoors/unexplored/rivers)
```

Comment:
This sucks. We can't generate the rivers (or chasms I guess) because they generate after the trees and stuff do, causing a LOT of floating objects. This is already a bug that's extremely difficult to fix, but it's especially egregious with this. 

Turning it off 'fixes' this, but also blocks rivers on lower-z's (call it variation or something)

Reply:
why not fix the order of the generation so it goes terrain, rivers, objects

Reply:
Terrain and objects are intertwined, so I’d need to tear them apart. I’ll uh, see how possible it is